### PR TITLE
fix: Replaced Clonedeep with Klona

### LINKED ISSUE #12936: [Task]: Replace cloneDeep with klona
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

As `klona` is faster when compared to lodash's cloneDeep. We would like to migrate from `cloneDeep` to `klona` in all the places where it is possible in client code. 

### Steps 
- [ ] Search for `cloneDeep` in client code
- [ ] Replace `cloneDeep` with `klona` 
- [ ] Add tests to verify you changes works as expected before and after the change.

### PR DESCRIPTION
**Pull request by LOKESH1076**
**PR fixes following issue(s)**
Fixes #12936

As klona is faster when compared to lodash's cloneDeep. We would like to migrate from cloneDeep to klona in all the places where it is possible in client code.
**Steps**

1.Search for cloneDeep in client code
2.Replace cloneDeep with klona
3.Test it to verify your changes works as expected before and after the change.

**Type of change**
Bug fix (non-breaking change which fixes an issue)

## Testing

**How Has This Been Tested?**
> Please describe the tests that you ran to verify your changes. Also list any relevant details for your test configuration.

- [x] Jest
- [x] Cypress
>
>

**## Checklist:**
**Dev activity**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

- **Refactor**
	- Replaced the usage of `lodash/cloneDeep` with the `klona` library for deep cloning objects across various components and utility functions. This change enhances performance and consistency in object cloning operations throughout the application.

### CODE REVIEW

**Comment by somangshu** [comment]
Something that has been attempted before unsuccessfully, @vsvamsi1 @rishabhrathod01 please take a call if this looks relevant

File: app/client/src/sagas/WidgetOperationSagas.tsx [hunk 1]
```diff
@@ -573,7 +574,7 @@ export function* batchUpdateWidgetDynamicPropertySaga(
 ) {
   const { updates, widgetId } = action.payload;
   const stateWidget: WidgetProps = yield select(getWidget, widgetId);
-  let widget = cloneDeep({ ...stateWidget });
+  let widget = klona({ ...stateWidget });
```
Comment (vsvamsi1) [inline app/client/src/sagas/WidgetOperationSagas.tsx:578]: We can remove the shallow copy here
```
cloneDeep(stateWidget)
```

File: app/client/src/widgets/JSONFormWidget/widget/propertyConfig/properties/common.ts [hunk 2]
```diff
@@ -65,9 +65,9 @@ function accessorValidation(
   const propertyPathChunks = propertyPath.split(".");
   const grandParentPath = propertyPathChunks.slice(0, -2).join(".");
   const schemaItemIdentifier = propertyPathChunks.slice(-2)[0]; // ['schema', '__root_field__', 'children', 'age', 'name'] -> age
-  const schema = lodash.cloneDeep(lodash.get(props, grandParentPath));
+  const schema = lodash.klona(lodash.get(props, grandParentPath));
   const RESTRICTED_KEYS = ["__array_item__", "__root_schema__"];
-  const currentSchemaItem = lodash.cloneDeep(schema[schemaItemIdentifier]);
+  const currentSchemaItem = lodash.klona(schema[schemaItemIdentifier]);
```
Comment (vsvamsi1) [inline app/client/src/widgets/JSONFormWidget/widget/propertyConfig/properties/common.ts:70]: This seems incorrect, klona does not come from lodash?

File: app/client/src/sagas/WidgetOperationSagas.tsx [hunk 3]
```diff
@@ -596,7 +597,7 @@ export function* setWidgetDynamicPropertySaga(
     widgetId,
   } = action.payload;
   const stateWidget: WidgetProps = yield select(getWidget, widgetId);
-  let widget = cloneDeep({ ...stateWidget });
+  let widget = klona({ ...stateWidget });
   const update = {
```
Comment (vsvamsi1) [inline app/client/src/sagas/WidgetOperationSagas.tsx:601]: I have a similar comment above related to this.

File: app/client/src/sagas/WidgetOperationSagas.tsx (hunk 1 above)
Comment (LOKESH1076) [reply app/client/src/sagas/WidgetOperationSagas.tsx:578]: ok @vsvamsi1 i will remove it

**Comment by vsvamsi1** [comment]
@LOKESH1076 
Can you profile our application and verify that you were able to see a drop in scripting.  
My experience with klona is that it performs similar to what cloneDeep does, its only when we go for either the /lite or /json version did i see massive improvement in performance. 
Also these libraries may not support certain types please check the documentation https://www.npmjs.com/package/klona to see if it maintains parity with cloneDeep.
cc @rishabhrathod01

**Pull request closed without merging**